src/main/java/org/apache/commons/lang3/time/FastDatePrinter.java

```diff
@@ -1118,7 +1118,7 @@ public void appendTo(final Appendable buffer, final Calendar calendar) throws IO
          */
         @Override
         public final void appendTo(final Appendable buffer, final int value) throws IOException {
-            appendDigits(buffer, value);
+            appendDigits(buffer, value % 100);
```

Comment:
I am not convinced by this because it patches the wrong spot.

Reply:
the name TwoDigitYearField implies this class encapsulates formatting of year field to a 2 digit string. as such, it seems reasonable for it to have the logic to take any year integer and map it to a decade followed by a year digits.
you can also see that the other method in this class 
`void appendTo(Appendable buf, Calendar calendar) throws IOException`
implementing the Rule interface, does the same

Reply:
@garydgregory WDYT?

Reply:
Hi @michael-o 
Earlier you said "I am not convinced by this because it patches the wrong spot." implying there was a "right" spot, but where would that be?

Reply:
I think here: https://issues.apache.org/jira/browse/LANG-1637?focusedCommentId=17257625&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17257625 and this method must throw IAE `if (value >= 100)`.

Reply:
Well... which way is this one going to go? @ugonen Are you going to update your PR or @michael-o do you want to provide a different PR?

Reply:
I think for the ease of use we have to use this one because my solution leaks abstraction. It makes too many assumptions about in the delegated `NumberRule`